scripts/run_lighthouse_tests_test.py

```diff
@@ -0,0 +1,301 @@
+        class MockFailedCompilerContextManager():
+            def __init__(self) -> None:
+                pass
+            def __enter__(self) -> MockFailedCompiler:
+                return MockFailedCompiler()
+            def __exit__(self, *unused_args: Any) -> None:
+                pass
+
+        def mock_failed_context_manager() -> MockFailedCompilerContextManager:
+            return MockFailedCompilerContextManager()
+        self.swap_webpack_compiler = self.swap_with_checks(
+            servers, 'managed_webpack_compiler', mock_failed_context_manager,
+            expected_args=(), expected_kwargs=[])
+        swap_isdir = self.swap_with_checks(
+            os.path, 'isdir', lambda _: False, expected_kwargs=[])
+
+        with self.print_swap, self.swap_webpack_compiler, swap_isdir:
+            with self.swap_sys_exit:
+                run_lighthouse_tests.run_webpack_compilation()
+
+        self.assertIn('Subprocess execution failed.', self.print_arr)
+
+    def test_run_lighthouse_checks_succesfully(self) -> None:
+        class MockTask:
+            returncode = 0
+            def communicate(self) -> tuple[bytes, bytes]:   # pylint: disable=missing-docstring
+                return (
+                    b'Task output',
+                    b'No error.')
+
+        def mock_popen(*unused_args: Any, **unsued_kwargs: Any) -> MockTask:  # pylint: disable=unused-argument
+            return MockTask()
+        swap_popen = self.swap_with_checks(
+            subprocess, 'Popen', mock_popen,
+            expected_args=((self.lighthouse_check_bash_command,),))
+
+        with self.print_swap, swap_popen:
+            run_lighthouse_tests.run_lighthouse_checks(
+                LIGHTHOUSE_MODE_PERFORMANCE, '1')
+
+        self.assertIn(
+            'Lighthouse checks completed successfully.', self.print_arr)
+
+    def test_run_lighthouse_checks_failed(self) -> None:
+        class MockTask:
+            returncode = 1
+            def communicate(self) -> tuple[bytes, bytes]:   # pylint: disable=missing-docstring
+                return (
+                    b'Task failed.',
+                    b'ABC error.')
+
+        def mock_popen(*unused_args: Any, **unsued_kwargs: Any) -> MockTask:  # pylint: disable=unused-argument
+            return MockTask()
+        swap_popen = self.swap_with_checks(
+            subprocess, 'Popen', mock_popen,
+            expected_args=((self.lighthouse_check_bash_command,),))
+
+        with self.print_swap, self.swap_sys_exit, swap_popen:
+            run_lighthouse_tests.run_lighthouse_checks(
+                LIGHTHOUSE_MODE_PERFORMANCE, '1')
+
+        self.assertIn('Return code: 1', self.print_arr)
+        self.assertIn('ABC error.', self.print_arr)
+        self.assertIn(
+            'Lighthouse checks failed. More details can be found above.',
+            self.print_arr)
+
+    def test_run_lighthouse_tests_in_accessibility_mode(self) -> None:
```

Comment:
Where are we asserting that the tests run in accessibility mode? Ditto for performance mode below

Reply:
Only way to distinguish between running in `accessibility` and `performance` mode is the arguments passed to `build.main()`. We are checking that by using `swap_with_checks` while creating a swap for `build.main()` method.

Reply:
Can you check based on what config file is passed here?

https://github.com/oppia/oppia/blob/7f35b33ef07c3a40630fa0417648553ea6f8d1ea/scripts/run_lighthouse_tests.py#L155

Reply:
Actually `subprocess.Popen()` is called twice when we call `run_lighthouse_tests.main()` method -- first in `run_lighthouse_puppeteer_script()` and then in `run_lighthouse_checks()` and both are calling different bash command. So we cannot use the same `swap_with_checks` for `subprocess.Popen` while calling the **main()** method. 

I am now checking that `run_lighthouse_checks()` method is called with correct parameters -- 'accessibility' or 'performance' accordingly. Since we have already made sure that `run_lighthouse_checks()` method works perfectly in the tests above, I think, this would do the work.